### ISSUE TITLE
Bug: ManaGunZ keeps building IRD file forever and can't cancel the operation.

### ISSUE THREAD

**Issue by foliage-burner** [Zarh/ManaGunZ#58]
Found another bug. I was building an ISO of the Ratchet and Clank Collection, and all was going well until the program started building the IRD. It didn't freeze, it didn't throw an error, but for some reason the expected time for building the IRD is increasing at the same rate as the elapsed time. There are no bytes being written from what I see on screen, and trying to cancel the operation brings up the cancelled notification, but ManaGunZ doesn't go back to the Blu-Ray disc menu.

Now, something I did beforehand was delete the 3Dump.bin file from tmp. I was testing a bug someone reported to me about building an ISO failing because the 3Dump.bin was invalid, or something along those lines. Could it be caused by that?

By the way, to be more clear, the IRD seemed to be building fine before. I believe the log stated that it was calculating the md5sum of various files. I think the error only occurred right at the end. I'll attach the video of the error occurring real time.

I'm using ManaGunZ v 1.40, with Evilnat 4.88

**Comment by foliage-burner** [comment]
Uploading VID_20210624_191810117.mp4…

**Comment by foliage-burner** [comment]
https://user-images.githubusercontent.com/83893307/123359212-bdce8f00-d521-11eb-8ece-25633f9e08fb.mp4

**Comment by foliage-burner** [comment]
Quitting ManaGunZ through XMB thankfully works, so I could escape through there without forcefully shutting off. The 3Dump.bin exists in the tmp folder, so I guess that wasn't the problem. I'll try dumping it again to see if this was a one time error (or maybe a problem with my external drive).

**Comment by Zarh** [comment]
The header is huge, i think it's a mistake I noticed last week. I already fixed it. I just uploaded the commit. Can you build the current version ? if no, I can upload a pkg to test it again.

**Comment by foliage-burner** [comment]
I haven't set up psl1ght yet, so if you could build a test pkg I'd really appreciate it (I could also try my hand at setting up a development environment right now, though it'll probably take me some time).

**Comment by Zarh** [comment]
I uploaded the pkg as a pre-release here : https://github.com/Zarh/ManaGunZ/releases/tag/1.41
Can you try ?

**Comment by foliage-burner** [comment]
Ok, it's building the decrypted ISO right now; it'll take a while to get to the IRD since the game is quite big.  I'll update in 20 minutes.

**Comment by foliage-burner** [comment]
https://user-images.githubusercontent.com/83893307/123450705-db3b4180-d591-11eb-9239-83e6a39e262f.mp4

**Comment by Zarh** [comment]
oh, sry I didn't try it. I'll fix it ASAP.

**Comment by Zarh** [comment]
Here it is : https://github.com/Zarh/ManaGunZ/releases/download/1.41/ManaGunZ_v1.40.pkg

It was an endianness mistake, sometimes I'm confused because I use the same code on windows (little endian) and the ps3 (big endian). I tried to write it properly, to not make the same mistake again....

**Comment by foliage-burner** [comment]
Ok, this time building the IRD works, and the bugs that happened with the first test build of 1.41 appear to be gone. 

One graphical hiccup though. While saving the ird, the predicted remaining time once again shows a large number in the hours section (like in the video). I almost thought I had encountered another bug, but after a while the screen went away and ManaGunZ put me back in the disc menu. My guess is that it was due to the size of the ird file. Might want to look into it later, but at least the core functionality is once again working.

By the way, does the latest (test) build include the changes that have to do with dumping iso and Blu-Ray keys to NTFS (w/ or w/o PS3ISO folder?). I can test that as well.
